gateway/src/main/java/io/zeebe/gateway/impl/broker/RequestRetryHandler.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.gateway.impl.broker;
+
+import io.zeebe.gateway.Loggers;
+import io.zeebe.gateway.cmd.BrokerErrorException;
+import io.zeebe.gateway.impl.broker.cluster.BrokerTopologyManager;
+import io.zeebe.gateway.impl.broker.request.BrokerRequest;
+import io.zeebe.gateway.impl.broker.response.BrokerResponse;
+import io.zeebe.protocol.record.ErrorCode;
+import java.net.ConnectException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class RequestRetryHandler {
+
+  private final BrokerClient brokerClient;
+  private final RoundRobinDispatchStrategy roundRobinDispatchStrategy;
+  private final BrokerTopologyManager topologyManager;
+
+  public RequestRetryHandler(
+      final BrokerClient brokerClient, final BrokerTopologyManager topologyManager) {
+    this.brokerClient = brokerClient;
+    roundRobinDispatchStrategy = new RoundRobinDispatchStrategy(topologyManager);
+    this.topologyManager = topologyManager;
+  }
+
+  public <BrokerResponseT> void sendRequest(
+      final BrokerRequest<BrokerResponseT> request,
+      final BrokerResponseConsumer<BrokerResponseT> responseConsumer,
+      final Consumer<Throwable> throwableConsumer) {
+    final Function<
+            BrokerRequest<BrokerResponseT>, CompletableFuture<BrokerResponse<BrokerResponseT>>>
+        requestSender = r -> brokerClient.sendRequest(r, false);
+    sendRequestInternal(request, requestSender, responseConsumer, throwableConsumer);
+  }
+
+  public <BrokerResponseT> void sendRequest(
+      final BrokerRequest<BrokerResponseT> request,
+      final BrokerResponseConsumer<BrokerResponseT> responseConsumer,
+      final Consumer<Throwable> throwableConsumer,
+      final Duration requestTimeout) {
+    final Function<
```

Comment:
It's used only internally in this class at two places. I don't think we can reuse the interface in BrokerRequestManager as it is operating at different abstraction.

Reply:
I know it's not used in the request manager, but I thought it might be reused there to avoid using boolean arguments (just an idea, not sure it'd be the best anyhow).

Also I see it used 4 times in this file? What did you mean by 2?

Reply:
Ya. There are 4. But it's all internal to this class.

Reply:
I find having specific types easier to understand, as function is pretty generic, but I get that Java doesn't have the concept of type aliasing, so it means creating a new interface/class. I'm not going to insist more if you feel strongly about it, as I don't have strong objective arguments :slightly_smiling_face: